compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -302,6 +302,12 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
 
         match maybe_transmutable {
             Answer::Yes => Ok(ImplSourceBuiltinData { nested: vec![] }),
+            Answer::No(reason) => Err(SelectionError::NotTransmutable {
+                reason,
+                scope,
+                src: type_at(1).skip_binder(),
```

Comment:
Do you need to return these types? I think you can extract them from the `FulfillmentError` during error reporting?

Reply:
Also, does this even need to be a separate `SelectionError`? I think you can just check if the [`obligation`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_infer/traits/struct.FulfillmentError.html#structfield.obligation) in the `FulfillmentError` is the transmutability trait, and extract the info again?

Reply:
is that the case? I don't understand how to get these fields from `PredicateObligation` -- do you have an example?

Reply:
Well you should have access to the trait predicate - https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_trait_selection/traits/error_reporting/mod.rs.html#653

Then from the trait predicate, you'd just be extracting the types from the substitutions just like the transmute trait confirmation code does.

Reply:
oh, gotcha, yeah that makes more sense now. I missed that `let trait_predicate = ...` line before.. will try it out

Reply:
Okay, I got it to work. I kept a separate `SelectionError` so that I can include the specific `NotTransmutable` error -- I don't think that is possible to have access to without either having the separate variant or calling rustc_transmute again.